File: GCDLib/UI/UtilityForms/frmInformation.vb
Namespace UI.UtilityForms
    Public Class frmInformation

        Private m_sFormTitle As String
        Private m_sMessage As String

        Public Sub InitializeSizeable(sTitle As String, sMessage As String, nMaxWidth As Integer, nMaxHeight As Integer)
            m_sFormTitle = sTitle
            m_sMessage = sMessage
            Me.MaximumSize = New System.Drawing.Size(nMaxWidth, nMaxHeight)
        End Sub

        Public Sub InitializeFixedDialog(sTitle As String, sMessage As String)
            m_sFormTitle = sTitle
            m_sMessage = sMessage
            Me.MaximizeBox = False
            Me.MinimizeBox = False
            Me.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Hide
            Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
        End Sub


        Private Sub InformationForm_Load(sender As Object, e As System.EventArgs) Handles Me.Load
            Me.Text = m_sFormTitle
            txtMessage.Text = m_sMessage
        End Sub
    End Class
End Namespace

File: GCDCore/UserInterface/UtilityForms/frmInformation.cs
using System;
using System.Windows.Forms;

namespace GCDCore.UserInterface.UtilityForms
{
    public partial class frmInformation : Form
    {
        private string m_sFormTitle;
        private string m_sMessage;

        public frmInformation()
        {
            InitializeComponent();
        }

        public void InitializeSizeable(string sTitle, string sMessage, int nMaxWidth, int nMaxHeight)
        {
            m_sFormTitle = sTitle;
            m_sMessage = sMessage;
            this.MaximumSize = new System.Drawing.Size(nMaxWidth, nMaxHeight);
        }

        public void InitializeFixedDialog(string sTitle, string sMessage)
        {
            m_sFormTitle = sTitle;
            m_sMessage = sMessage;
            this.MaximizeBox = false;
            this.MinimizeBox = false;
            this.SizeGripStyle = System.Windows.Forms.SizeGripStyle.Hide;
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
        }

        private void InformationForm_Load(object sender, System.EventArgs e)
        {
            this.Text = m_sFormTitle;
            txtMessage.Text = m_sMessage;
        }
    }
}